aten/src/ATen/native/cuda/LinearAlgebra.cu

```diff
@@ -50,6 +28,31 @@ Tensor prepare_matrix_for_cublas(Tensor& tensor, bool& transpose_tensor) {
   return tensor_;
 }
 
+Tensor prepare_batch_matrix_for_cublas(Tensor& tensor, bool& transpose_tensor, bool transpose_result, int64_t& ld_tensor) {
```

Comment:
Instead of adding a `prepare_batch_matrix_for_cublas`, can we just slightly modify `prepare_matrix_for_cublas` into:

```C++
Tensor prepare_matrix_for_cublas(Tensor& tensor, bool& transpose_tensor, int batch_dims=0) {
  Tensor tensor_;
  IntArrayRef tensor_strides = tensor.strides();
  IntArrayRef tensor_sizes = tensor.sizes();

  if ((tensor_strides[batch_dims] == 1) && (tensor_strides[1 + batch_dims] >= std::max<int64_t>(1, tensor_sizes[batch_dims]))) {
    tensor_ = tensor;
    transpose_tensor = false;
  } else if ((tensor_strides[1 + batch_dims] == 1) && (tensor_strides[batch_dims] >= std::max<int64_t>(1, tensor_sizes[1 + batch_dims]))) {
    tensor_ = tensor;
    transpose_tensor = true;
  } else {
    transpose_tensor = true;
    tensor_ = tensor.clone(at::MemoryFormat::Contiguous);
  }

  return tensor_;
}
```
and reuse it?

Reply:
I don't think so because `prepare_matrix_for_cublas` doesn't use `transpose_result` to decide which strides should be used in the conditional statements